reframe/core/systems.py

```diff
@@ -13,6 +13,169 @@
 from reframe.core.environments import (Environment, ProgEnvironment)
 
 
+class Processor(jsonext.JSONSerializable):
+    '''A representation of a processor inside ReFrame.
+
+    .. warning::
+       Users may not create :class:`Processor` objects directly.
+    '''
+
+    def __init__(self, processor_info):
+        self._arch = None
+        self._num_cpus = None
+        self._num_cpus_per_core = None
+        self._num_cpus_per_socket = None
+        self._num_sockets = None
+        self._topology = None
+        self._info = processor_info
+
```

Comment:
can we add `num_numa_nodes` and `num_cores_per_numa_node` ?

Reply:
Yes, I think these make sense as calculated properties and we should perhaps add more like the cpu lists from the different cpu sets.